iphone/cli/hooks/run.js

```diff
@@ -13,119 +13,212 @@ const appc = require('node-appc'),
 	ioslib = require('ioslib'),
 	i18n = appc.i18n(__dirname),
 	__ = i18n.__,
-	__n = i18n.__n;
+	__n = i18n.__n,
+	net = require('net'),
+	exec = require('child_process').exec;
 
 exports.cliVersion = '>=3.2';
 
 exports.init = function (logger, config, cli) {
 	cli.addHook('build.post.compile', {
 		priority: 10000,
 		post: function (builder, finished) {
-			if (cli.argv.target !== 'simulator') {
-				return finished();
-			}
-
-			if (cli.argv['build-only']) {
-				logger.info(__('Performed build only, skipping running of the application'));
-				return finished();
-			}
-
-			logger.info(__('Launching iOS Simulator'));
-
 			let simStarted = false,
-				lastLogger = 'debug';
-			const startLogTxt = __('Start simulator log'),
-				endLogTxt = __('End simulator log'),
-				levels = logger.getLevels(),
+				lastLogger = 'debug',
+				startLogTxt = '',
+				endLogTxt = '';
+
+			const levels = logger.getLevels(),
 				trimRE = new RegExp('^.*' + builder.tiapp.name + '\\[[^\\]]+\\]\\s*', 'g'),
 				logLevelRE = new RegExp('^(\u001b\\[\\d+m)?\\[?(' + levels.join('|') + '|log|timestamp)\\]?\\s*(\u001b\\[\\d+m)?(.*)', 'i');
 
+			if (cli.argv.target === 'macos') {
+				// TO DO: It does force quit. Find another way to quit.
+
+				const quitCommand = 'pkill -QUIT -x ' + builder.tiapp.name;
+
+				exec(quitCommand, function () {
+
+					logger.info(__('Launching Mac application'));
+					const command = 'open -a ' + builder.iosBuildDir + '/' + builder.tiapp.name + '.app/Contents/MacOS/' + builder.tiapp.name;
+
+					exec(command, function () {
+					// TO DO: It only log app messages.
+					// Need to  make logging better for macos and should be comparable to logging as in ios simulator and device.
+					// Probably can  be done after first release.
+						startLogTxt = __('Start mac application log');
+						endLogTxt = __('End mac application log');
+						function connectToLogServer() {
+							if (builder.tiLogServerPort) {
+								logger.trace(__('Trying to connect to log server port %s...', builder.tiLogServerPort).grey);
+
+								(function tryConnecting() {
+									var client = net.connect(builder.tiLogServerPort, function () {
+										logger.trace(__('Connected to log server port %s...', builder.tiLogServerPort).grey);
+
+										function disconnectLogServer() {
```

Comment:
<!--
  0 failure: 
  1 warning:  iphone/cli/hooks/...
  
  
  DangerID: danger-id-Danger;
  File: iphone/cli/hooks/run.js;
  Line: 59;
-->

- :warning: iphone/cli/hooks/run.js line 59 – 'disconnectLogServer' is defined but never used. (no-unused-vars)